DistributionSystemApi/MailLibrary/SMTPMailService.cs

```diff
@@ -0,0 +1,85 @@
+﻿using System.Net.Mail;
+using System.ComponentModel.DataAnnotations;
+using DistributionSystemApi.MailLibrary;
+
+namespace MailLibrary
+{
+    public class SMTPMailService : IMailService<MailModel>
+    {
+        private const string InvalidSMTPClientConfExceptionMessage = "Check configuration parameters";
+
+        private const string InvalidEmailsCountExceptionMessage = "Check mails count";
+
+        private readonly SmtpClient _smptClient;
+        private readonly IMailValidationService _mailValidationService;
+
+        public SMTPMailService(SmtpClient smtpClient, IMailValidationService mailValidationService)
+        {
+            _smptClient = smtpClient ?? throw new ArgumentNullException(nameof(smtpClient));
+
+            _mailValidationService = mailValidationService ?? throw new ArgumentNullException(nameof(mailValidationService));
+        }
+
+        public async Task SendEmailAsync(MailModel mail, CancellationToken cancellationToken)
+        {
+            _mailValidationService.ValidateMailAndThrowError(mail);
+
+            try
+            {
+                using (MailMessage mailMessage = new MailMessage())
+                {
+                    Map(mail, mailMessage);
+
+                    await _smptClient.SendMailAsync(mailMessage, cancellationToken);
+                }
+            }
+            catch (InvalidOperationException ex)
+            {
+                throw new InvalidOperationException(InvalidSMTPClientConfExceptionMessage, ex);
+            }
+        }
+
+        public async Task SendEmailsAsync(IEnumerable<MailModel> mails, CancellationToken cancellationToken)
+        {
+            if(mails == null)
+            {
+                throw new ArgumentException(InvalidEmailsCountExceptionMessage);
+            }
+
+            foreach (var mail in mails)
+            {
+                await SendEmailAsync(mail, cancellationToken);
+            }
+        }
+
+        private void Map(MailModel mail, MailMessage mailMessage)
+        {
+
```

Comment:
better create MailMessage object in this method using MailModel object as source